File: app/src/androidTest/java/ru/org/adons/cblock/app/TestComponent.java
package ru.org.adons.cblock.app;

import dagger.Component;
import ru.org.adons.cblock.datamodel.BlockListModelTest;
import ru.org.adons.cblock.datamodel.CallLogModelTest;
import ru.org.adons.cblock.ui.base.ViewScope;

/**
 * Provide Application and data-model dependencies for test
 */

@ViewScope
@Component(dependencies = ApplicationComponent.class, modules = TestModule.class)
public interface TestComponent {

    void inject(CallLogModelTest test);

    void inject(BlockListModelTest test);

}

File: app/src/androidTest/java/ru/org/adons/cblock/app/TestComponent.kt
package ru.org.adons.cblock.app

import dagger.Component
import ru.org.adons.cblock.data.BlockListModelTest
import ru.org.adons.cblock.data.CallLogModelTest
import ru.org.adons.cblock.scope.ViewScope

/**
 * Provide Application and data-model dependencies for test
 */
@ViewScope
@Component(dependencies = arrayOf(ApplicationComponent::class))
interface TestComponent {

    fun inject(test: CallLogModelTest)

    fun inject(test: BlockListModelTest)

}
